services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/gdpr/GdprService.kt

```diff
@@ -1,26 +1,86 @@
 package fi.hel.haitaton.hanke.gdpr
 
+import fi.hel.haitaton.hanke.allu.CustomerType
 import fi.hel.haitaton.hanke.application.Application
 import fi.hel.haitaton.hanke.application.ApplicationService
+import fi.hel.haitaton.hanke.domain.YhteystietoTyyppi
+import fi.hel.haitaton.hanke.permissions.HankekayttajaRepository
+import fi.hel.haitaton.hanke.permissions.PermissionRepository
 import mu.KotlinLogging
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
 import org.springframework.stereotype.Service
 import org.springframework.transaction.annotation.Transactional
 
 private val logger = KotlinLogging.logger {}
 
+interface GdprService {
+    @Transactional(readOnly = true) fun findGdprInfo(userId: String): CollectionNode?
+
+    @Transactional(readOnly = true) fun findApplicationsToDelete(userId: String): List<Application>
+
+    @Transactional fun deleteApplications(applicationsToDelete: List<Application>, userId: String)
+}
+
+@Service
+@ConditionalOnProperty(
+    name = ["haitaton.features.user-management"],
+    havingValue = "true",
+    matchIfMissing = false
+)
```

Comment:
What do you mean by more uniform? The `matchIfMissing` value?

Reply:
My mistake, I got confused. But yes, perhaps `matchIfMissing=false` should be everywhere where this property is used?

Reply:
It would be nice if you could offer some reasoning behind this suggestion.

If we change everything as you suggest and the property is missing for some reason, the application won't start. Using `matchIfMissing=true` means that that's the implementation we want the application to use, ie. the default.

Though, I don't think the property can be missing, since there's a default value in the application properties. I think of `matchIfMissing` as a redundancy that can save the application on a bad day. However, it might be preferable that the application doesn't start in a case where the property has been dropped from the application properties. Fail-fast and all that.

Reply:
I wasn't clear enough, sorry.
What I meant is that now there are multiple places where property `haitaton.features.user-management` is checked. Some places there are also `matchIfMissing` but not everywhere. E.g. `HakemusController` doesn't have it. And in that GdprService there are `havingValue = true` and `matchIfMissing = false` and in ApplicationController there is `havingValue = false` and `matchIfMissing = true`. It might be more clear if these different settings for the same property could be more uniform. For those classes, that depend on `haitaton.features.user-management` to be `true` there would be only `havingValue = true` (as `matchIfMissing` is `false` by default) and those classes that can work without the property or it having `false` as value could have `havingValue = false` and `matchIfMissing = true`. How does this sound to you?

Reply:
So the actual change would be to remove `matchIfMissing = false` from this class's annotation, because the default is false anyway?

Reply:
Or add it to other places where this property is used.

Reply:
Well, of course I can add it to HakemusController because I created that Controller in the first place. So, never mind about all this.